app/src/main/java/com/example/awesomelanguagelearning/chooseLanguage/ui/ChooseLanguageScreen.kt

```diff
@@ -35,59 +37,89 @@ fun ChooseLanguageScreen(
     val viewModel: ChooseLanguageViewModel = koinViewModel()
 
     val state by viewModel.state.collectAsState()
-    val pagerSize = state.pages.size
     val currentPageIndex = state.currentPage
-    val pagerState = rememberPagerState { pagerSize }
+    val pagerState = rememberPagerState { state.pages.size }
 
     LaunchedEffect(currentPageIndex) {
         pagerState.animateScrollToPage(currentPageIndex)
     }
 
+    ChooseLanguageContent(
+        pagerState = pagerState,
+        screenState = state,
+        currentPageIndex = currentPageIndex,
+        navigateToNextScreen = navigateToNextScreen,
+        navigateBackFlow = navigateBack,
+        navigateBackScreen = viewModel::onBack,
+        onNextClick = viewModel::onNext,
+        onItemClick = viewModel::onListItemClick
+    )
+}
+
+
+@OptIn(ExperimentalMaterial3Api::class, ExperimentalFoundationApi::class)
+@Composable
+private fun ChooseLanguageContent(
+    pagerState: PagerState,
+    screenState: ChooseLanguageState,
+    currentPageIndex: Int = 0,
+    navigateToNextScreen: () -> Unit = {},
+    navigateBackFlow: () -> Unit = {},
+    navigateBackScreen: () -> Unit = {},
+    onNextClick: () -> Unit = {},
+    onItemClick: (UUID) -> Unit = {}
+) {
     Scaffold(
         modifier = Modifier.fillMaxSize(),
         topBar = {
             Toolbar(
-                text = stringResource(R.string.complited_title, currentPageIndex + 1, pagerSize),
+                text = stringResource(
+                    R.string.complited_title,
+                    currentPageIndex + 1,
```

Comment:
@igel911 do you really need to pass `currentPageIndex` if you pass the whole `pagerState` as a parameter?

Reply:
@slothdeboss Well, you're right. It can be removed but I'll miss it so much